src/net-utils.ts

```diff
@@ -39,6 +39,16 @@ async function isPortOpen(
   });
 }
 
+async function isTerminalClosed(terminal: vscode.Terminal): Promise<boolean> {
```

Comment:
I think the verb tense in the function name is off here; we're watching for a future closing of the terminal rather than testing if a given terminal is in the closed state. How would you name this function?

Reply:
Past patterns I've used have been `getTerminalClosedPromise`, `waitTerminalClose`, and `joinTerminal` (a la [pthreads](https://pubs.opengroup.org/onlinepubs/009695399/functions/pthread_join.html))

Reply:
I like and went with `getTerminalClosedPromise()`